sound/soc/sof/ipc4-pcm.c

```diff
@@ -107,7 +215,29 @@ static int sof_ipc4_trigger_pipelines(struct snd_soc_component *component,
 		pipeline->state = state;
 	}
 
-	return ret;
+	if (ret < 0)
+		return ret;
+
+	if (state == SOF_IPC4_PIPE_RUNNING) {
+		u32 abi_version;
+		u32 abi_offset;
+
+		abi_offset = offsetof(struct sof_ipc4_fw_registers, abi_ver);
+		sof_mailbox_read(sdev, sdev->fw_info_box.offset + abi_offset, &abi_version,
+				 sizeof(abi_version));
+
+		if (abi_version != SOF_IPC4_FW_REGS_ABI_VER) {
+			dev_err(sdev->dev, "unsupported abi version %d for time stamp\n",
+				abi_version);
+			return -EINVAL;
+		}
+
+		ret = sof_ipc4_build_time_info(sdev, &spcm->stream[substream->stream]);
+		if (ret < 0)
+			return ret;
+	}
```

Comment:
It would be good to have comments to explain why this if (state == SOF_IPC4_PIPE_RUNNING) is needed and what happens if it's not true? I think you've provided details already but I have lost all context here.